### PR TITLE
Detail page feedback form

### PR DESCRIPTION
**Pull request by Coreball**
### Summary 

This pull request implements the feedback form on the place detail view. When the user presses the "is this accurate?" button used for the old Google Form, a new feedback view controller popup appears and the user can enter feedback. For this feedback form, this includes: is this accurate yes/no, user-observed crowdedness with Flux density pills, a wait time picker, and a text field for any additional comments. When the user reaches the end of the popup, they can submit the feedback, which sends that data along with the current eatery and Flux-predicted crowdedness to the back end.

- [x] implemented feedback form popup for places detail view
  - [x] "is this accurate?" yes/no
  - [x] "how crowded?" density pills
  - [x] "how many minutes?" wait time picker (buckets of 2 mins)
  - [x] "anything else?" comments text field
  - [x] "thank you" page, auto dismissing after 5s
- [x] implemented basic integration with Flux API

### Next Steps

- [x] branching behavior to skip density pills if user said it was accurate
- [ ] grey out buttons and prevent progression when nothing pressed (possibly)
- [ ] refine/verify backend integration

### Test Plan 

Screen Shot 2020-12-14 at 19 27 02
Screen Shot 2020-12-14 at 19 29 23
Screen Shot 2020-12-14 at 19 29 28
Screen Shot 2020-12-14 at 19 29 37
Screen Shot 2020-12-14 at 19 29 42
Screen Shot 2020-12-14 at 19 29 48

### Notes 

Customizable popups and radio buttons are not native iOS, so new components had to be created to satisfy the designs. They work, but aren't as elegant as their Apple-designed cousins. Instead of presenting a new view controller in the modal iOS fashion or navigating to a new page, it is hidden and shown with a black background.

### CODE REVIEW

**Comment by AnshG714** [comment]
Very well done Changyuan!!! I particularly like how you've broken up different feedback questions using delegation patterns, makes the code really modular. I've just pushed a commit that allows for branching of the questions, and I've also added a default value for the feedback accuracy radio button. We need to decide exactly how we're gonna provide default values to the users/show them an error message if we choose to not let users move forward if they haven't explicitly filled out a value in the feedback form (this is what I'd prefer). Either way, we can merge this in right now, and pick up next semester/over winter break. Super impressed!!!

**Review by AnshG714 — APPROVED** [review]

**Comment by bungaepyo** [comment]
Great job! I like how you added the questions as subviews to the background and programmatically navigate through them. I have Z E R O knowledge with swift and ios so please take that into account. I don't know how long usually `UIViewController` are, but `FeedbackViewController` looks somewhat too long. I think, if possible, it would help to process some of the UI in a different module. Also, I left a comment regarding the waittimes.

File: Campus Density/FeedbackQuestions/WaitTimeQuestion.swift [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+import UIKit
+
+protocol WaitTimeQuestionDelegate: class {
+    func waitTimeWasChanged(waitTime: Int)
+}
+
+class WaitTimeQuestion: FeedbackQuestion, UIPickerViewDataSource, UIPickerViewDelegate {
+    weak var delegate: WaitTimeQuestionDelegate?
+    let options = ["0-2", "2-4", "4-6", "6-8", "8-10", "10-12", "12-14", "14-16", "16-18", "18-20", "20+"]
```
Comment (bungaepyo) [inline Campus Density/FeedbackQuestions/WaitTimeQuestion.swift:17]: I like how you added more granularity to the waittime options I should do that too. However, is it necessary to include `"16-18", "18-20", "20+"`?

Comment (Coreball) [reply Campus Density/FeedbackQuestions/WaitTimeQuestion.swift:17]: Not sure, I think we just chose buckets of two minutes. I agree that those buckets on the high end seem less meaningful as 14 and 16 minutes don't feel as different as 2 and 4 minutes. We should probably standardize them later so the data coming into the backend has the same source options.

**Pull request merged**